engine/src/main/junit-shared/org/operaton/bpm/engine/test/util/JobExecutorHelper.java

```diff
@@ -0,0 +1,320 @@
+  }
+
+  /**
+   * Waits for a specified condition to be met within a given time period, repeatedly checking the condition
+   * at specified intervals. If the condition is not met within the time limit, a {@link ProcessEngineException}
+   * is thrown.
+   *
+   * @param condition       A {@link Callable} returning a {@link Boolean}, representing the condition to be evaluated.
+   *                        The method will wait until this condition evaluates to {@code true}.
+   * @param maxMillisToWait The maximum amount of time in milliseconds to wait for the condition to be satisfied.
+   * @param checkInterval   The interval in milliseconds at which the condition will be re-evaluated.
+   *                        If this value is greater than {@code maxMillisToWait}, it will be adjusted to match {@code maxMillisToWait}.
+   */
+  public static void waitForCondition(Callable<Boolean> condition, long maxMillisToWait, long checkInterval) {
+    if (maxMillisToWait < checkInterval) {
+      checkInterval = maxMillisToWait;
+    }
+    try {
+      await().atMost(maxMillisToWait, TimeUnit.MILLISECONDS)
+          .pollInterval(checkInterval, TimeUnit.MILLISECONDS)
+          .ignoreExceptions() // In case condition throws an exception during polling
+          .until(condition);
+    } catch (Exception e) {
+      throw new ProcessEngineException("Time limit of " + maxMillisToWait + " was exceeded.");
+    }
+  }
+
+  /**
+   * Waits for the job executor to process all jobs associated with a specific process instance.
+   * This method repeatedly checks for job completion at specified intervals within a defined maximum wait time.
+   * Throws a {@link ProcessEngineException} in case the timeout is exceeded.
+   * Optionally shuts down the job executor after processing if specified.
+   *
+   * @param processInstanceId the ID of the process instance for which jobs need to be processed
+   * @param maxMillisToWait   the maximum time in milliseconds to wait for all jobs to be processed
+   * @param checkInterval     the interval in milliseconds to check for job completion
+   * @param jobExecutor       the {@link JobExecutor} responsible for processing the jobs
+   * @param managementService the {@link ManagementService} used to interact with jobs management
+   * @param shutdown          a flag indicating whether to shut down the job executor after processing
+   */
+  public static void waitForJobExecutorToProcessAllJobs(String processInstanceId,
+                                                        long maxMillisToWait,
+                                                        long checkInterval,
+                                                        JobExecutor jobExecutor,
+                                                        ManagementService managementService,
+                                                        boolean shutdown) {
+    activateJobExecutor(jobExecutor);
```

Comment:
Maybe return a boolean by activateJobExecutor when the executor was activated and shutdown only in that case? Can the method argument maybe obsolete then? Why should clients want to indicate a shutdown?